File: LeetCode/PalindromeNumber.cpp
#include <iostream>

using namespace std;

//获取一半长度的整数转化成int，与另一半的整数进行比较
bool isPalindrome(int x) 
{
	if (x < 0 || (x!=0 && x%10==0))//负数不为回文数或者结尾为0且该数不为0整数的也不为回文
		return false;

	int halfInt = 0;

	while (halfInt < x)
	{
		halfInt *= 10;
		halfInt += x % 10;
		x = x / 10;
	}

	if (x == halfInt)
		return true;
	else
	{
		if (halfInt / 10 == x)//处理12321,在上面的情况下halfInt会变成123,而x会变成12
			return true;
		else
			return false;
	}
}

int main()
{
	if (isPalindrome(1))
		cout << "Yes";
	else
		cout << "No";

	return 0;
}

File: LeetCode/iter1/python/PalindromeNumber.py
class Solution:
    # @param {integer} x
    # @return {boolean}
    def isPalindrome(self, x):
        if(x>=0 and x<10):#个位数是回文数
            return True
        
        if(x<0 or x % 10 == 0):
            return False
            
        latterHalf = 0#记录后半段的整数
        while(x>latterHalf):
            #print("x:",x," latterHalf:",latterHalf)
            digit = x%10
            #print("digit:",digit)
            x = x//10
           # print("x:",x)
            latterHalf = latterHalf*10+digit
            #print("latterHalf:",latterHalf)

        if(x == latterHalf):
            return True
                
        latterHalf //= 10
        
        if(x==latterHalf):
            return True
        else:
            return False
            
s = Solution()
if(s.isPalindrome(1)):
    print("Yes")
else:
    print("No")